core/src/main/java/org/elasticsearch/indices/IndexingMemoryController.java

```diff
@@ -50,22 +52,37 @@
-    public static final String INDEX_BUFFER_SIZE_SETTING = "indices.memory.index_buffer_size";
+    public static final Setting<String> INDEX_BUFFER_SIZE_SETTING = new Setting<String>("indices.memory.index_buffer_size", (s) -> "10%",
+                                                                                        (s) -> {
+                                                                                            if (s.endsWith("%")) {
+                                                                                                try {
+                                                                                                    Double.parseDouble(s.substring(0, s.length() - 1));
+                                                                                                } catch (NumberFormatException nfe) {
+                                                                                                    throw new IllegalArgumentException("unknown value for [indices.memory.index_buffer_size]: must be X% or a size value (e.g. XMB) but was: " + s);
+                                                                                                }
+                                                                                            } else {
+                                                                                                try {
+                                                                                                    ByteSizeValue.parseBytesSizeValue(s, "indices.memory.index_buffer_size");
+                                                                                                } catch (Throwable t) {
+                                                                                                    throw new IllegalArgumentException("unknown value for [indices.memory.index_buffer_size]: must be X% or a size value (e.g. XMB) but " + t.getMessage());
+                                                                                                }
+                                                                                            }
+
+                                                                                            return s;
+                                                                                        },
+                                                                                        Property.NodeScope);
 
     /** Only applies when <code>indices.memory.index_buffer_size</code> is a %, to set a floor on the actual size in bytes (default: 48 MB). */
-    public static final String MIN_INDEX_BUFFER_SIZE_SETTING = "indices.memory.min_index_buffer_size";
+    public static final Setting<ByteSizeValue> MIN_INDEX_BUFFER_SIZE_SETTING = Setting.byteSizeSetting("indices.memory.min_index_buffer_size", new ByteSizeValue(48, ByteSizeUnit.MB), Property.NodeScope);
```

Comment:
I wonder if it would be better to use the form of `byteSizeSetting` that lets us set a minimum value so that we can reject negative byte size settings?